src/main/java/dev/cdevents/CDEventTypes.java

```diff
@@ -14,217 +18,922 @@ private CDEventTypes() {
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("name", name)
+                        .withExtension("url", url)
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery environment modified event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param name
+     * @param url
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * environment created event
+     */
+
+    public static CloudEvent createEnvironmentModifiedEvent(
+            final String eventType, final String id,
+            final URI source, final String name,
+            final String url, final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("name", name)
+                        .withExtension("url", url)
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery environment deleted event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param name
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * environment deleted event
+     */
+
+    public static CloudEvent createEnvironmentDeletedEvent(
+            final String eventType, final String id,
+            final URI source, final String name,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("name", name)
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery service deployed event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param environment
+     * @param artifactId
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * service deployed event
+     */
+
+    public static CloudEvent createServiceDeployedEvent(
+            final String eventType, final String id,
+            final Environment environment, final URI artifactId,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("environment",  environment.toString())

```

Comment:
please use jackson-databind to write this `environment` as String, say objectMapper.writeValueAsString(environment)